src/v0/sources/shopify/util.js

```diff
@@ -145,6 +144,40 @@ const getAnonymousIdFromDb = async (message, metricMetadata) => {
   }
   return getAnonymousId(message);
 };
+/**
+ * This function sets the sessionId based on cart_token or id from the properties of message.
+ * @param {*} message
+ * @returns
+ */
+const getSessionIdFromDB = async (message, metricMetadata) => {
+  const cartToken = getCartToken(message);
+  const { event } = message;
+  if (!isDefinedAndNotNull(cartToken)) {
+    return null;
+  }
+  let sessionId;
+  try {
+    sessionId = await RedisDB.getVal(`${cartToken}`, 'sessionId');
+  } catch (e) {
+    logger.debug(`Time: ${new Date()} {{SHOPIFY::}} sessionId get call Failed due redis error ${e}`);
+
+    stats.increment('shopify_redis_failures', {
+      type: 'get',
+      ...metricMetadata,
+    });
+  }
+  stats.increment('shopify_redis_calls', {
+    type: 'get',
+    ...metricMetadata,
+  });
+  if (sessionId === null) {
```

Comment:
I think `sessionId` can be `undefined` as well not only `null`, so should we change this condition ?

Reply:
Also please verify what would be value of `sessionId` in `catch` block. I think it could be `undefined` & handle accordingly

Reply:
If there is no data for a particular key then redis return `null` and not `undefined` and that why its handled in this format